sonic_installer/bootloader/grub.py

```diff
@@ -85,35 +85,31 @@ def remove_image(self, image):
         run_command('grub-set-default --boot-directory=' + HOST_PATH + ' 0')
         click.echo('Image removed')
 
-    def verify_image_platform(self, image_path):
-        if not os.path.isfile(image_path):
-            return False
-
-        # Get running platform's ASIC
+    def platform_in_platforms_asic(self, platform, image_path):
+        """
+        For those images that don't have devices list builtin, check_output will raise an exception.
+        In this case, we simply return True to make it worked compatible as before.
+        Otherwise, we need to check if platform is inside the supported target platforms list.
+        """
         try:
-            version_info = device_info.get_sonic_version_info()
-            if version_info:
-                asic_type = version_info['asic_type']
-            else:
-                asic_type = None
-        except (KeyError, TypeError) as e:
-            click.echo("Caught an exception: " + str(e))
+            with open(os.devnull, 'w') as fnull:
+                p = subprocess.Popen(["sed", "-e", "1,/^exit_marker$/d", image_path], stdout=subprocess.PIPE, preexec_fn=default_sigpipe)
+                output = subprocess.check_output(["tar", "xf", "-", PLATFORMS_ASIC, -O], stdin=p.stdout, stderr=fnull, preexec_fn=default_sigpipe, text=True)
+                return platform in output
```

Comment:
>output [](http://example.com/codeflow?start=35&length=6)

`output` could be a long list. You can reduce the memory consuption by continuing pipe and grep, or use line-by-line file manipulation.

Reply:
I was thinking using pipe and grep. But seems Popen dont have better ways to handle the case of unexisted file. 
Because 'platform_asic nonexistence' and 'platform unmatch' will both produce the empty string if using pipe and grep.
I am thinking of  Popen(['sed'...) first then check_call(['tar'...) first to see if there is any CalledProcessError to confirm platforms_asic exist. Then do the above sed & tar then pipe and grep.
But it will cost more time as you can see. Do you prefer the pipe and grep over check_output?